website/bot.py

```diff
@@ -0,0 +1,126 @@
+from pathlib import Path
+
+import openai
+from dotenv import find_dotenv, load_dotenv
+from langchain.chains import ConversationalRetrievalChain
+from langchain.memory import ConversationSummaryMemory
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain_community.document_loaders import (
+    DirectoryLoader,
+    Docx2txtLoader,
+    PyPDFLoader,
+    TextLoader,
+    UnstructuredMarkdownLoader,
+)
+from langchain_community.vectorstores import FAISS
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_openai import ChatOpenAI, OpenAIEmbeddings
+from openai import OpenAI
+
+load_dotenv(find_dotenv(), override=True)
+
+
+def is_api_key_valid(api_key):
+    client = OpenAI(api_key=api_key)
+    try:
+        response = client.completions.create(
+            prompt="Hello", model="gpt-3.5-turbo-instruct", max_tokens=1
+        )
+        return True
+    except openai.APIConnectionError as e:
+        print("Failed to connect to OpenAI API: {e}")
```

Comment:
The `e` is unused. I believe you meant print(f"...") here and below

Reply:
@arkid15r , make another commit

Reply:
`e` was acceptable. if you want to use unique names they need to comply with the snake case naming convention.

Reply:
@arkid15r  done